x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/find_rules_status_route.ts

```diff
@@ -46,37 +49,59 @@ export const findRulesStatusesRoute = (router: IRouter) => {
         }
     */
       try {
+        const failingRules = await findRules({
+          alertsClient,
+          perPage: ids.length,
+          page: undefined,
+          sortField: undefined,
+          sortOrder: undefined,
+          filter: 'alert.attributes.executionStatus.status: "error"',
+          fields: ['executionStatus'],
+        });
+
         const ruleStatusClient = ruleStatusSavedObjectsClientFactory(savedObjectsClient);
-        const statuses = await body.ids.reduce<Promise<RuleStatusResponse | {}>>(
-          async (acc, id) => {
-            const lastFiveErrorsForId = await ruleStatusClient.find({
-              perPage: 6,
-              sortField: 'statusDate',
-              sortOrder: 'desc',
-              search: id,
-              searchFields: ['alertId'],
-            });
-            const accumulated = await acc;
 
-            // Array accessors can result in undefined but
-            // this is not represented in typescript for some reason,
-            // https://github.com/Microsoft/TypeScript/issues/11122
-            const currentStatus = convertToSnakeCase<IRuleStatusAttributes>(
-              lastFiveErrorsForId.saved_objects[0]?.attributes
+        await Promise.all(
```

Comment:
Wait...Should there be something to the left of this `await`?

Is this just an oopsie here or am I missing something? This looks like it's doing a map and returning transforms but nothing to the left of the map is doing anything?

Reply:
This part:

```ts
        const failingRules = await findRules({
          alertsClient,
          perPage: ids.length,
          page: undefined,
          sortField: undefined,
          sortOrder: undefined,
          filter: 'alert.attributes.executionStatus.status: "error"',
          fields: ['executionStatus'],
        });
```

Looks off because you're getting per page the number of id's you have but not all the id's you send in are going to be in failure state. And then you only look at that first page of errors below?

Reply:
This might be related but I was able to sometimes get this error message popping up when I went to the monitoring page and caused a lot of errors by turning on and off a single rule:
 
<img width="824" alt="Screen Shot 2020-10-05 at 2 34 55 PM" src="https://user-images.githubusercontent.com/1151048/95129372-28cc0180-0718-11eb-9753-2e02382d9f69.png">

Reply:
> Wait...Should there be something to the left of this await?
Nope this is just writing the statuses. I pull them back out when querying for them later on in that reduce function.

As for the `findRules` piece I'm not sure paging comes into play here unless the saved objects client has a maximum for how many items it can return per page? The only thing I saw in the saved objects client was it defaults to 20 documents per page, which as long as I'm passing in something in the `perPage` param I should be in the clear there.

Reply:
I see what you're saying about it merging in the status during a find with those functions.


Here though, don't you need to add a filter to get the id's you passed in?

```ts
```ts
const failingRules = await findRules({
  alertsClient,
  perPage: ids.length,
  page: undefined,
  sortField: undefined,
  sortOrder: undefined,
  filter: 'alert.attributes.executionStatus.status: "error"',
  fields: ['executionStatus'],
});
```

Don't you need to construct something like this with the id's passed in `filter: 'alert.attributes.executionStatus.status: "error" AND (alert.attributes.id: id_1 or alert.attributes.id: id_2 or alert.attributes.id: id_3)'`?

It looks like you're not filtering by id?

Reply:
Also, it looks like on each page load when it hits the route it is re-writing the last rule status of error and erasing the first 5 eventually if it finds the failingRules.

You can see this by causing an error that sets the `alert.attributes.executionStatus.status` and then going to that rule's page and re-loading the page 5 times quickly. It will keep adding a new entry to the last 5 and rotate the others out.

For example, this rule has a `alert.attributes.executionStatus.status` of error and I reload the page 5 times quickly and now it thinks it experienced the same error within seconds of each other (and recently) when in fact the rule is turned off and it keeps merging in the last failure with a new timestamp on page loads:

Showing me I have experienced 5 errors when I haven't run it in the last 30 minutes, I just reloaded the page: 
<img width="1218" alt="Screen Shot 2020-10-05 at 6 05 25 PM" src="https://user-images.githubusercontent.com/1151048/95144294-8ae72f80-0735-11eb-8d2f-0c4fc818d858.png">

It also constantly thinks that it ran a few seconds ago on each page load when in fact it hasn't. This looks like it is just updating its last run and error structure on each page load. 
<img width="1218" alt="Screen Shot 2020-10-05 at 6 06 35 PM" src="https://user-images.githubusercontent.com/1151048/95144338-a0f4f000-0735-11eb-8741-331d15b29cd7.png">